src/legacy/core_plugins/data/public/search/search_bar/components/saved_query_manager.tsx

```diff
@@ -0,0 +1,360 @@
+    };
+    if (isOpen) {
+      fetchQueries();
+    }
+  }, [isOpen]);
+
+  const goToPage = (pageNumber: number) => {
+    setActivePage(pageNumber);
+  };
+
+  const savedQueryDescriptionText = i18n.translate(
+    'data.search.searchBar.savedQueryDescriptionText',
+    {
+      defaultMessage:
+        'Saved queries allow you to save common search snippets and filters for later use.',
+    }
+  );
+
+  const noSavedQueriesDescriptionText =
+    i18n.translate('data.search.searchBar.savedQueryNoSavedQueriesText', {
+      defaultMessage: 'There are no saved queries.',
+    }) +
+    ' ' +
+    savedQueryDescriptionText;
+
+  const savedQueryPopoverTitleText = i18n.translate(
+    'data.search.searchBar.savedQueryPopoverTitleText',
+    {
+      defaultMessage: 'Saved Queries',
+    }
+  );
+
+  const onDeleteSavedQuery = (savedQuery: SavedQuery) => {
+    setSavedQueries(savedQueries.filter(currentSavedQuery => currentSavedQuery !== savedQuery));
+
+    if (loadedSavedQuery && loadedSavedQuery.id === savedQuery.id) {
+      onClearSavedQuery();
+    }
+
+    deleteSavedQuery(savedQuery.id);
+  };
+
+  const savedQueryPopoverButton = (
+    <EuiButtonEmpty
+      className="euiFormControlLayout__prepend"
+      iconType="arrowDown"
+      iconSide="right"
+      onClick={() => {
+        setIsOpen(!isOpen);
+      }}
+      aria-label={i18n.translate('data.search.searchBar.savedQueryPopoverButtonText', {
+        defaultMessage: 'See saved queries',
+      })}
+      data-test-subj="saved-query-manager-popover-button"
+    >
+      #
+    </EuiButtonEmpty>
+  );
+
+  const savedQueryRows = () => {
+    const savedQueriesDisplayRows = savedQueries.slice(
+      activePage * pageCount,
+      activePage * pageCount + pageCount
+    );
+    return savedQueriesDisplayRows.map(savedQuery => (
+      <li key={savedQuery.id}>
+        <EuiFlexGroup justifyContent="spaceBetween" gutterSize="none">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              onClick={() => {
+                onLoad(savedQuery);
+                setIsOpen(false);
+              }}
+              flush="left"
+              data-test-subj={`load-saved-query-${savedQuery.attributes.title}-button`}
+            >
+              <EuiScreenReaderOnly>
+                <span>
+                  {i18n.translate('data.search.searchBar.savedQueryScreenReaderOpenText', {
+                    defaultMessage: 'Open saved query',
+                  })}
+                </span>
+              </EuiScreenReaderOnly>
+              {savedQuery.attributes.title}
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup gutterSize="none" justifyContent="flexEnd" alignItems="center">
+              <EuiFlexItem>
+                {savedQuery.attributes.description && (
+                  <EuiIconTip type="iInCircle" content={savedQuery.attributes.description} />
```

Comment:
@myasonik we just added this tooltip. I checked it in VoiceOver and it read out the contents, but is there anything else special we should do to make it accessible?

Reply:
It's already usable as is so that's awesome. (Thanks EUI!) But, we can make this better by giving it a unique `aria-label` instead of relying on the default "Info".

```suggestion
<EuiIconTip 
  type="iInCircle" 
  content={savedQuery.attributes.description} 
  aria-label=`${savedQuery.attributes.title} description`
/>
```

Reply:
👍 ffe3a13